barretenberg/cpp/src/barretenberg/vm2/simulation_helper.cpp

```diff
@@ -32,6 +30,9 @@ struct ProvingSettings {
     using BytecodeRetrievalEventEmitter = EventEmitter<BytecodeRetrievalEvent>;
     using BytecodeHashingEventEmitter = EventEmitter<BytecodeHashingEvent>;
     using BytecodeDecompositionEventEmitter = EventEmitter<BytecodeDecompositionEvent>;
+    using AddressDerivationEventEmitter = EventEmitter<AddressDerivationEvent>;
+    using ClassIdDerivationEventEmitter = EventEmitter<ClassIdDerivationEvent>;
+    using SiloingEventEmitter = EventEmitter<SiloingEvent>;
```

Comment:
This is already kind of growing a lot. I might change it to a default vs Noop emitter later.